### PR TITLE
#596 #597  Create Series navigation

### LINKED ISSUE #596: Create links from Category page to the Series one
We have no mockups for it. Use your own design solution

### LINKED ISSUE #597: Create link from Product page to the Series one
We have no mockups for it. Use your own design solution

### PR DESCRIPTION
**Pull request by duker33**
Closes #596
Closes #597

### CODE REVIEW

File: stroyprombeton/models.py [hunk 1]
```diff
@@ -47,6 +47,30 @@ def get_absolute_url(self):
...
+                    Option.objects
+                    .select_related('product')
+                    .select_related('product__category')
+                    .filter(
+                        product__category__in=(
+                            Category.objects
+                            .select_related('parent')
+                            .prefetch_related('children')
+                            .filter(id=self.id)
+                            .get_descendants(include_self=True)
```
Comment (artemiy312) [inline stroyprombeton/models.py]:
```suggestion
                            self.get_descendants(include_self=True)
```
https://django-mptt.readthedocs.io/en/latest/models.html#get-descendants-include-self-false

It seems you can use only this instead `Category.objects.select_related()...`
Probably it should make it a single query

File: stroyprombeton/models.py [hunk 2]
```diff
@@ -263,6 +287,12 @@ def get_siblings(self, offset):
             .select_related('page')[:offset]
         )
 
+    def get_series(self) -> models.QuerySet:
+        product = Product.objects.prefetch_related('options').get(id=self.id)
+        return Series.objects.filter(options__in=product.options.active())
```
[the remark this answers is not in the document]
Comment (duker33) [reply stroyprombeton/models.py]: @artemiy312 , i tried it. In this case i get AttributeError. That's the reason i lookup product object (inside a Product's method, he-he) at the previous string

File: stroyprombeton/models.py [hunk 3]
```diff
@@ -263,6 +287,12 @@ def get_siblings(self, offset):
             .select_related('page')[:offset]
         )
 
+    def get_series(self) -> models.QuerySet:
```
[the remark this answers is not in the document]
Comment (duker33) [reply stroyprombeton/models.py]: @artemiy312 , sure, i'll do it.
Good arch is the best documentation, so i'll subtask creating `ProductQS.get_series()` method with good performance

File: stroyprombeton/models.py (hunk 1 above)
Comment (duker33) [reply stroyprombeton/models.py]: i'll subtask this investigation

**Pull request merged**